src/Tvheadend.h

```diff
@@ -386,7 +407,10 @@ class CTvheadend
   const tvheadend::Settings   m_settings;
   
   CHTSPConnection             m_conn;
-  CHTSPDemuxer                m_dmx;
+
+  std::vector<CHTSPDemuxer*>  m_dmx;
+  int                         m_dmx_active;
```

Comment:
I'd store a pointer to the actual demuxer instead of the vector index, that way all the loops over `m_dmx` can be changed to `for (auto *dmx : m_dmx)` which is a bit more concise.